### PR TITLE
[INTERNAL] [Client encryption]: Add baseline benchmarks for `Microsoft.Azure.Cosmos.Encryption.Custom`

### PR DESCRIPTION
**Pull request by juraj-blazek**
# Pull Request Template

## Description

Adding a `Microsoft.Azure.Cosmos.Encryption.Custom.Performance.Tests` with `Encrypt`/`Decrypt` benchmarks for different document size. Attaching also initial baseline report with the current memory allocations.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Closing issues

Contributes to #4678

### CODE REVIEW

File: Microsoft.Azure.Cosmos.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.29123.88
+# Visual Studio Version 17
+VisualStudioVersion = 17.12.35209.166
```
Comment (juraj-blazek) [inline Microsoft.Azure.Cosmos.sln:4]: Is there a need to stick with older VS version? This can be reverted if needed.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.sln:4]: For now we are sticking with the older version, can you please revert this file.

**Review by kirankumarkolli — APPROVED** [review]

**Review by kr-santosh — APPROVED** [review]

**Comment by juraj-blazek** [comment]
@kirankumarkolli could you have a look, too, please. thanks

**Review by kr-santosh — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**